ts/Extensions/MouseWheelZoom/MouseWheelZoom.ts

```diff
@@ -66,198 +70,256 @@ const optionsToObject = (
-        if (inner.height > outer.height) {
-            inner.height = outer.height;
-            inner.y = outer.y;
-        } else {
-            inner.y = outer.y + outer.height - inner.height;
-        }
-    }
-    if (inner.height > outer.height) {
-        inner.height = outer.height;
-    }
-    if (inner.y < outer.y) {
-        inner.y = outer.y;
-    }
-
-    return inner;
+    return {
+        rangeStart: innerStart,
+        rangeWidth: innerWidth
+    };
 };
 
 let wheelTimer: number,
     startOnTick: boolean|undefined,
     endOnTick: boolean|undefined;
 
 /**
+ * Temporarly disable `axis.startOnTick` and `axis.endOnTick` to allow zooming
+ * for small values.
  * @private
- */
-const zoomBy = function (
+*/
+const waitForAutomaticExtremes = function (
+    axis: Axis
+) : void {
+    const axisOptions = axis.options;
+
+    // Options interfering with yAxis zoom by setExtremes() returning
+    // integers by default.
+    if (defined(wheelTimer)) {
+        clearTimeout(wheelTimer);
+    }
+
+    if (!defined(startOnTick)) {
+        startOnTick = axisOptions.startOnTick;
+        endOnTick = axisOptions.endOnTick;
+    }
+
+    // Temporarily disable start and end on tick, because they would
+    // prevent small increments of zooming.
+    if (startOnTick || endOnTick) {
+        axisOptions.startOnTick = false;
+        axisOptions.endOnTick = false;
+    }
+    wheelTimer = setTimeout((): void => {
+        if (defined(startOnTick) && defined(endOnTick)) {
+            // Repeat merge after the wheel zoom is finished, #19178
+            axisOptions.startOnTick = startOnTick;
+            axisOptions.endOnTick = endOnTick;
+
+            // Set the extremes to the same as they already are, but now
+            // with the original startOnTick and endOnTick. We need
+            // `forceRedraw` otherwise it will detect that the values
+            // haven't changed. We do not use a simple yAxis.update()
+            // because it will destroy the ticks and prevent animation.
+            const { min, max } = axis.getExtremes();
+            axis.forceRedraw = true;
+            axis.setExtremes(min, max);
+            startOnTick = endOnTick = void 0;
+        }
+    }, 400);
+};
+
+/**
+* Calulate the ratio of mouse position on the axis to it's length. If mousePos
+* doesn't exist, returns 0.5;
+* @private
+*/
+const getMouseAxisRatio = function (
     chart: Chart,
-    howMuch: number,
-    centerXArg: number,
-    centerYArg: number,
-    mouseX: number,
-    mouseY: number,
-    options: MouseWheelZoomOptions
-): boolean {
-    const xAxis = chart.xAxis[0],
-        yAxis = chart.yAxis[0],
-        yOptions = yAxis.options,
-        type = pick(options.type, chart.zooming.type, ''),
-        zoomX = /x/.test(type),
-        zoomY = /y/.test(type);
+    axis: Axis,
+    mousePos: number
+) : number {
+    const mouseAxisRatio = mousePos ? ((mousePos - axis.pos) / axis.len) : 0.5,
```

Comment:
this check will return false, if `mousePos` is 0, and I think this is not a desired behaviour :D 
for the check use `U.defined` method, and if the mousePos can be undefined, let's reflect it in the parameter type either by making it optional, or by piping it with the `undefined` type.

Reply:
Btw, it can be simplified to just 
```suggestion
   if (!defined(mousePos)) {
   return 0.5;
   }
    const mouseAxisRatio = ((mousePos - axis.pos) / axis.len);
```

Reply:
Good catch, thanks!